test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js

```diff
@@ -3,865 +3,267 @@
-      const canonicalFunctionsPrefix = `${
-        awsPackage.serverless.service.service
-      }-${awsPackage.provider.getStage()}`;
-
-      expect(
-        awsPackage.serverless.service.provider.compiledCloudFormationTemplate.Resources[
-          awsPackage.provider.naming.getRoleLogicalId()
-        ]
-      ).to.deep.equal({
-        Type: 'AWS::IAM::Role',
-        Properties: {
-          AssumeRolePolicyDocument: {
-            Version: '2012-10-17',
-            Statement: [
-              {
-                Effect: 'Allow',
-                Principal: {
-                  Service: ['lambda.amazonaws.com'],
-                },
-                Action: ['sts:AssumeRole'],
-              },
-            ],
-          },
-          Path: '/',
-          Policies: [
-            {
-              PolicyName: {
-                'Fn::Join': [
-                  '-',
-                  [awsPackage.serverless.service.service, awsPackage.provider.getStage(), 'lambda'],
-                ],
-              },
-              PolicyDocument: {
-                Version: '2012-10-17',
-                Statement: [
-                  {
-                    Effect: 'Allow',
-                    Action: ['logs:CreateLogStream', 'logs:CreateLogGroup'],
-                    Resource: [
-                      {
-                        'Fn::Sub':
-                          'arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:' +
-                          `log-group:/aws/lambda/${canonicalFunctionsPrefix}*:*`,
-                      },
-                    ],
-                  },
-                  {
-                    Effect: 'Allow',
-                    Action: ['logs:PutLogEvents'],
-                    Resource: [
-                      {
-                        'Fn::Sub':
-                          'arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:' +
-                          `log-group:/aws/lambda/${canonicalFunctionsPrefix}*:*:*`,
-                      },
-                    ],
-                  },
-                ],
-              },
-            },
-          ],
-          RoleName: {
-            'Fn::Join': [
-              '-',
-              [
-                awsPackage.serverless.service.service,
-                awsPackage.provider.getStage(),
-                {
-                  Ref: 'AWS::Region',
-                },
-                'lambdaRole',
-              ],
-            ],
-          },
-        },
+describe('lib/plugins/aws/package/lib/mergeIamTemplates.test.js', () => {
+  describe('No default role', () => {
+    it('should not create role resource if there are no functions', async () => {
+      const { cfTemplate, awsNaming } = await runServerless({
+        fixture: 'aws',
+        cliArgs: ['package'],
       });
+      const iamRoleLambdaExecution = awsNaming.getRoleLogicalId();
+      const resourceIam = cfTemplate.Resources[iamRoleLambdaExecution];
+      expect(resourceIam).to.be.undefined;
```

Comment:
Let's reconfigure to `expect(cfTemplate.Resources).to.not.have.property(iamRoleLambdaExecution)`, it's more up to the point